src/merlin.ml

```diff
@@ -49,17 +49,16 @@ let dot_merlin sctx ~dir ~scope ({ requires; flags; _ } as t) =
       >>^ (fun (libs, flags) ->
         let ppx_flags = ppx_flags sctx ~dir ~scope ~src_dir:remaindir t in
         let internals, externals =
-          List.fold_left libs ~init:([], []) ~f:(fun (internals, externals) ->
-            function
-            | Lib.Internal (path, lib) ->
-              let spath =
-                Path.drop_optional_build_context path
-                |> Path.reach ~from:remaindir
-              in
-              let bpath = Path.reach (Lib.lib_obj_dir path lib) ~from:remaindir in
+          List.fold_left libs ~init:([], []) ~f:(fun (internals, externals) (lib : Lib.t) ->
+            match Lib.public_name lib with
+            | Some pkg ->
+              internals, ("PKG " ^ pkg) :: externals
```

Comment:
That doesn't seem right to me, local libraries can have a public name and we shouldn't use `PKG` for those, given that `PKG` is for findlib packages. In fact, I wonder if we ever need to use `PKG` at all, given that we already know the transitive closure we could just generate `S` and `B` directives directly even for external libraries